packages/endpoint/endpoint.js

```diff
@@ -0,0 +1,154 @@
+import fs from 'fs';
+import axios from 'axios';
+import RestroomCore from "@restroom-mw/core";
+const { Restroom } = RestroomCore;
+
+const ACTIONS = {
+  EXTERNAL_CONNECTION: "that I have an endpoint named {}",
+  EXTERNAL_OUTPUT: "I connect to {} and save the output",
+  PASS_OUTPUT: "pass the output to {}"
+};
+
+export default (req, res, next) => {
+  const rr = new Restroom(req, res);
+  let externalSourceKeys = [];
+  let externalSourceValue = [];
+
+  let outputName = 'output';
+  let outputValue;
+
+  //get the api call from restroom that contains username and contract name
+  const originalUrl = rr._req.originalUrl;
+  const userContractDir = originalUrl.split('/api/')[1];
+
+  //Get the directories for keys and data
+  const completeDir = process.cwd() + '/contracts/' + userContractDir;
+  const keyFile = completeDir + '.keys';
+  const dataFile = completeDir + '.data';
+
+  rr.onBefore(async (params) => {
+
+    const { zencode, data } = params;
+
+    if (zencode.match(ACTIONS.EXTERNAL_CONNECTION)) {
+      externalSourceKeys = zencode.paramsOf(ACTIONS.EXTERNAL_CONNECTION);
+    }
+
+    if (zencode.match(ACTIONS.EXTERNAL_OUTPUT)) {
+
+      if (externalSourceKeys.length > 0) {
+
+        //if exists get the value for the external source in the keys or data files that have the externalSourceKey key
+        if (fs.existsSync(keyFile)) {
+          const keyFileContent = JSON.parse(fs.readFileSync(keyFile).toString());
+          for (const key of externalSourceKeys) {
+            // use function getValueForKey to recursively search the json for the key and get the value
+            const value = getValueForKey(keyFileContent, key);
+            if (value) {
+              try {
+                // make the api call with url in value
+                const response = await axios.get(value);
+                outputValue = response.data;
+              } catch (error) {
+                outputValue = null;
+              }
+              rr.setData(outputName, outputValue);
+            }
+          };
+        }
+
+
+        if (fs.existsSync(dataFile)) {
+          const dataFileContent = JSON.parse(fs.readFileSync(dataFile).toString());
```

Comment:
In restroom-mw `data` are passed via the request.body, but `conf` and `keys` are read from the filesystem. Reading data from the filesystem seems to be specific for your purpose only